### ISSUE TITLE
Support "strict" mapping option

### ISSUE THREAD

**Issue by hlascelles** [SteveSanderson/knockout.mapping#154]
We use knockout to great effect, and have all our models fully predefined (since we do a lot of calculations on the client side). However, we run into problems with new GUI releases connecting to old service layers, or using old cached data, thus:

```
    var model = {name: {first: ko.observable(), last: ko.observable()}};
    ko.mapping.fromJS({name: 'Bob Green'}, {}, model); // Note different model structure
    console.log(model.name.first()); // Kaboom!  TypeError: Object Bob Green has no method 'first' 
```

In this example we find out pretty quickly what has happened, but sometimes it is minutes later when we try and retrieve the value from an observable we expect to be there, but isn't. 

The solution to the above problem is, of course, to upgrade the service layer and invalidate the server cache, but only once you've figured out why observables are vanishing.

Fail Fast is my mantra - can we add a "strict" mapping option to blow up in this scenario? eg:

```
    var model = {name: {first: ko.observable(), last: ko.observable()}};
    ko.mapping.fromJS({name: 'Bob Green'}, {strict: true}, model); // Knockout Mapping Error! 'name' is not a leaf property.
```

Similarly, for properties that don't exist:

```
    var model = {name: {first: ko.observable(), last: ko.observable()}};
    ko.mapping.fromJS({age: '25'}, {strict: true}, model); // Knockout Mapping Error! 'age' is not a known property.
```

We have some code that does this (a spin round with _ko.mapping.visitModel_), but it feels inefficient to traverse the model twice, and doesn't handle arrays well.

It might be as simple as checking for _mappedRootObject_ being _undefined_ in the function _updateViewModel_, but there may be a faster/more efficient way.

**Comment by sagacity** [comment]
Yes, this should be possible to add in precisely the way you proposed. If you want to give it a shot, feel free to do so. :) Otherwise it'll be a while, I'm afraid.

**Comment by vicary** [comment]
How about adding a verbose option, and do `console.warn()` instead? Throwing only blame the first encounter, by warning we can probably work on more than one new mapping at a time.
